overturemaps/core.py

```diff
@@ -5,6 +5,13 @@
 import pyarrow.dataset as ds
 import pyarrow.fs as fs
 
+# Allows for optional import of additional dependencies
+try: 
+    import geopandas as gpd
+    from geopandas import GeoDataFrame
+except ImportError: 
+    gpd = None
+    GeoDataFrame = None
```

Comment:
Maybe here just add a variable like `HAS_GEOPANDAS` which you can check later rather than setting gpd and GeoDataFrame to None

Reply:
This is the pattern I saw for optional dependencies, we need to at least set it for `GeoDataFrame` as it's the return type, but I can set a flag for checking in the function instead.

Reply:
Ah that makes sense for GeoDataFrame